lib/main.dart

```diff
@@ -61,57 +60,64 @@ const schulCloudAppConfig = AppConfig(
 );
 
 Future<void> main({AppConfig appConfig = schulCloudAppConfig}) async {
-  Logger.level = Level.debug;
-  logger
-    ..i('Starting…')
-    ..d('Initializing hive…');
-  await initializeHive();
+  WidgetsFlutterBinding.ensureInitialized();
+  await runWithErrorReporting(() async {
+    Logger.level = Level.debug;
+    logger
+      ..i('Starting…')
+      ..d('Registering first services…');
+    // We register these first as they're required for error reporting.
+    services
+      ..registerSingleton(appConfig)
+      ..registerSingletonAsync((_) => StorageService.create());
 
-  logger.d('Initializing services…');
-  services
-    ..registerSingletonAsync((_) async {
-      // We need to initialize TimeMachine before launching the app, and using
-      // get_it to keep track of initialization statuses is the simplest way.
-      // Hence we just ignore the return value.
-      var timeZone = await FlutterNativeTimezone.getLocalTimezone();
-      if (timeZone == 'GMT') {
-        timeZone = 'UTC';
-      }
-      await TimeMachine.initialize({
-        'rootBundle': rootBundle,
-        'timeZone': timeZone,
-      });
-    }, instanceName: 'ignored')
-    ..registerSingleton(appConfig)
-    ..registerSingletonAsync((_) => StorageService.create())
-    ..registerSingleton(SnackBarService())
-    ..registerSingleton(NetworkService())
-    ..registerSingleton(ApiNetworkService())
-    ..registerSingleton(FileService())
-    ..registerSingletonAsync((_) => DeepLinkingService.create())
-    ..registerSingleton(CalendarBloc())
-    ..registerSingleton(SignInBloc());
+    logger.d('Initializing hive…');
+    await initializeHive();
 
-  logger.d('Adding custom licenses to registry…');
-  LicenseRegistry.addLicense(() async* {
-    yield EmptyStateLicense();
-  });
-
-  logger.d('Running…');
-  runApp(
-    FutureBuilder<void>(
-      future: services.allReady(),
-      builder: (_, snapshot) {
-        if (!snapshot.hasData) {
-          return Container(
-            color: Colors.white,
-            alignment: Alignment.center,
-            child: CircularProgressIndicator(),
-          );
+    logger.d('Registering remaining services…');
+    services
+      ..registerSingletonAsync((_) async {
+        // We need to initialize TimeMachine before launching the app, and using
+        // get_it to keep track of initialization statuses is the simplest way.
```

Comment:
```suggestion
        // GetIt to keep track of initialization statuses is the simplest way.
```
same as in `error_reporting.dart`